desired_state_generator/src/gen_desired_state.py

```diff
@@ -0,0 +1,205 @@
+def parse_vehicle_signal_interface(config: Dict[str, Any]) -> List[str]:
+    """Parses the vehicle signal interface config.
+
+    Args:
+        config: The json-config of the interface, as defined in the appManifest.json.
+
+    Returns:
+        List[str]: A list of requirements defined by the config.
+    """
+    requirements = []
+    src = str(config["src"])
+    vss_release_prefix = (
+        "https://github.com/COVESA/vehicle_signal_specification/releases/download/"
+    )
+    version = ""
+    if vss_release_prefix in src:
+        version = src.removeprefix(vss_release_prefix).split("/")[0]
+        requirements.append(f"vss-source-default-vss:{version}")
+    elif is_uri(src):
+        version = get_md5_from_uri(src)
+        requirements.append(f"vss-source-custom-vss:{version}")
+    else:
+        version = get_md5_for_file(src)
+        requirements.append(f"vss-source-custom-vss:{version}")
+
+    requirements.append(f"data-broker-grpc:{get_package_version('vehicledatabroker')}")
+
+    datapoints = config["datapoints"]["required"]
+    for datapoint in datapoints:
+        path = str(datapoint["path"]).lower().replace(".", "-")
+        access = datapoint["access"]
+        requirements.append(f"vss-{access}-{path}:{version}")
+
+    return requirements
+
+
+def parse_grpc_interface(config: Dict[str, Any]) -> str:
+    """Parses the grpc interface config.
+
+    Args:
+        config: The json-config of the interface, as defined in the appManifest.json.
+
+    Returns:
+        str: The requirement with md5-hash of the proto-file as version.
+    """
+    src = str(config["src"])
+
+    return f"dataprovider-proto-grpc:{get_md5_from_uri(src)}"
+
+
+def get_md5_from_uri(src: str) -> str:
+    """Get the md5-hash of a file defined by an URI.
+
+    Args:
+        str: The URI of the file.
+
+    Returns:
+        str: The md5-hash of the file.
+    """
+    md5 = hashlib.md5(usedforsecurity=False)
+    with requests.get(src, timeout=30) as source:
+        for chunk in source.iter_content(chunk_size=4096):
+            md5.update(chunk)
+
+    return md5.hexdigest()
+
+
+def get_md5_for_file(file_path: str):
+    """Get the md5-hash of a local file defined by a path.
+
+    Args:
+        str: The local path to the file.
+
+    Returns:
+        str: The md5-hash of the file.
+    """
+    md5 = hashlib.md5(usedforsecurity=False)
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            md5.update(chunk)
+
+    return md5.hexdigest()
+
+
+def get_package_version(package_id: str) -> str:
+    """Get the version of the used mqtt-broker.
+
+    Returns:
+        str: The version of the mqtt-broker defined in the runtime.json.
+    """
+    config = velocitas_lib.services.get_specific_service(package_id).config
+    version = config.image.split(":")[-1]
+    return version
+
+
+def parse_interfaces(interfaces: List[Dict[str, Any]]) -> List[str]:
+    """Parses the defined interfaces.
```

Comment:
Should be imperative, pydocstyle should complain.
```suggestion
    """Parse the defined interfaces.
```